### DISCUSSION TITLE
Translations

### DISCUSSION THREAD

**Discussion by KurtisLiggett** [snappysnappydog/Simple-IP-Config#132] [General]
### How to use language files

Available language files are automatically installed with the default installation.

To manually install new language files:
1. Download one of the available translations.
2. Copy the .JSON file into the "lang" directory.
       a.  For most installations, this is "C:\Program Files (x86)\Simple IP Config\lang"
       b.  For the portable version, this is the lang directory within the directory where the main exe is located
3. Open the program and go to Tools->Settings
a. Available language files will be listed in the dropdown menu
b. Select the language and click the Save button
c. The program will be restarted in the selected language

Translations Available:
* [English](https://github.com/KurtisLiggett/Simple-IP-Config/blob/master/lang/lang-en-US.json)
* [Italian](https://github.com/KurtisLiggett/Simple-IP-Config/blob/master/lang/lang-it-IT.json) - thanks @bovirus!
* [Russian](https://github.com/KurtisLiggett/Simple-IP-Config/blob/master/lang/lang-ru-RU.json) - thanks @mrkaban!
* [Spanish](https://github.com/KurtisLiggett/Simple-IP-Config/blob/master/lang/lang-es-ES.json) - thanks @andago9!
* [French](https://github.com/KurtisLiggett/Simple-IP-Config/blob/master/lang/lang-fr-FR.json) - thanks @logounet!
* [German](https://github.com/KurtisLiggett/Simple-IP-Config/blob/master/lang/lang-de-DE.json) - thanks @uDEV2019!
* [Romanian](https://github.com/KurtisLiggett/Simple-IP-Config/blob/master/lang/lang-ro-RO.json) - thanks @ms74ro!
* [Dutch](https://github.com/KurtisLiggett/Simple-IP-Config/blob/master/lang/lang-nl-NL.json) - thanks @PvdE75!
* [Chinese](https://github.com/KurtisLiggett/Simple-IP-Config/blob/master/lang/lang-zh-CN.json) - thanks @gooduc!

### How to contribute a new language file
Translation contributions are always welcome.
To create a new language file:
1. Download one of the available files above to use as the template
2. Save the file as "lang-{Country code}.json" where {Country code} is the code listed in the table below
3. Translate all of the strings in the language file
4. Change the _name_ property to the name of the language, as it should be shown in the program selection list
5. Change the _code_ property to the country code (below) for the new language
6. Fork and clone the repository and submit a pull request to the _master_ branch with the new file, or just post the new file in this discussion.

<details>
  <summary>Click to Expand Language Code Table</summary>

| **Hex** | **Country code** | **Description**                              |
|:-------:|:----------------:|----------------------------------------------|
|   0004  |      zh-CHS      | Chinese - Simplified                         |
|   0401  |       ar-SA      | Arabic - Saudi Arabia                        |
|   0402  |       bg-BG      | Bulgarian - Bulgaria                         |
|   0403  |       ca-ES      | Catalan - Spain                              |
|   0404  |       zh-TW      | Chinese (Traditional) - Taiwan               |
|   0405  |       cs-CZ      | Czech - Czech Republic                       |
|   0406  |       da-DK      | Danish - Denmark                             |
|   0407  |       de-DE      | German - Germany                             |
|   0408  |       el-GR      | Greek - Greece                               |
|   0409  |       en-US      | English - United States                      |
|   040A  |   es-ES_tradnl   | Spanish - Spain                              |
|   040B  |       fi-FI      | Finnish - Finland                            |
|   040C  |       fr-FR      | French - France                              |
|   040D  |       he-IL      | Hebrew - Israel                              |
|   040E  |       hu-HU      | Hungarian - Hungary                          |
|   040F  |       is-IS      | Icelandic - Iceland                          |
|   0410  |       it-IT      | Italian - Italy                              |
|   0411  |       ja-JP      | Japanese - Japan                             |
|   0412  |       ko-KR      | Korean - Korea                               |
|   0413  |       nl-NL      | Dutch - Netherlands                          |
|   0414  |       nb-NO      | Norwegian (Bokmål) - Norway                  |
|   0415  |       pl-PL      | Polish - Poland                              |
|   0416  |       pt-BR      | Portuguese - Brazil                          |
|   0417  |       rm-CH      | Romansh - Switzerland                        |
|   0418  |       ro-RO      | Romanian - Romania                           |
|   0419  |       ru-RU      | Russian - Russia                             |
|   041A  |       hr-HR      | Croatian - Croatia                           |
|   041B  |       sk-SK      | Slovak - Slovakia                            |
|   041C  |       sq-AL      | Albanian - Albania                           |
|   041D  |       sv-SE      | Swedish - Sweden                             |
|   041E  |       th-TH      | Thai - Thailand                              |
|   041F  |       tr-TR      | Turkish - Turkey                             |
|   0420  |       ur-PK      | Urdu - Pakistan                              |
|   0421  |       id-ID      | Indonesian - Indonesia                       |
|   0422  |       uk-UA      | Ukrainian - Ukraine                          |
|   0423  |       be-BY      | Belarusian - Belarus                         |
|   0424  |       sl-SI      | Slovenian - Slovenia                         |
|   0425  |       et-EE      | Estonian - Estonia                           |
|   0426  |       lv-LV      | Latvian - Latvia                             |
|   0427  |       lt-LT      | Lithuanian - Lithuanian                      |
|   0428  |    tg-Cyrl-TJ    | Tajik (Cyrillic) - Tajikistan                |
|   0429  |       fa-IR      | Persian - Iran                               |
|   042A  |       vi-VN      | Vietnamese - Vietnam                         |
|   042B  |       hy-AM      | Armenian - Armenia                           |
|   042C  |    az-Latn-AZ    | Azeri (Latin) - Azerbaijan                   |
|   042D  |       eu-ES      | Basque - Basque                              |
|   042E  |      hsb-DE      | Upper Sorbian - Germany                      |
|   042F  |       mk-MK      | Macedonian - Macedonia                       |
|   0432  |       tn-ZA      | Setswana / Tswana - South Africa             |
|   0434  |       xh-ZA      | isiXhosa - South Africa                      |
|   0435  |       zu-ZA      | isiZulu - South Africa                       |
|   0436  |       af-ZA      | Afrikaans - South Africa                     |
|   0437  |       ka-GE      | Georgian - Georgia                           |
|   0438  |       fo-FO      | Faroese - Faroe Islands                      |
|   0439  |       hi-IN      | Hindi - India                                |
|   043A  |       mt-MT      | Maltese - Malta                              |
|   043B  |       se-NO      | Sami (Northern) - Norway                     |
|   043e  |       ms-MY      | Malay - Malaysia                             |
|   043F  |       kk-KZ      | Kazakh - Kazakhstan                          |
|   0440  |       ky-KG      | Kyrgyz - Kyrgyzstan                          |
|   0441  |       sw-KE      | Swahili - Kenya                              |
|   0442  |       tk-TM      | Turkmen - Turkmenistan                       |
|   0443  |    uz-Latn-UZ    | Uzbek (Latin) - Uzbekistan                   |
|   0444  |       tt-RU      | Tatar - Russia                               |
|   0445  |       bn-IN      | Bangla - Bangladesh                          |
|   0446  |       pa-IN      | Punjabi - India                              |
|   0447  |       gu-IN      | Gujarati - India                             |
|   0448  |       or-IN      | Oriya - India                                |
|   0449  |       ta-IN      | Tamil - India                                |
|   044A  |       te-IN      | Telugu - India                               |
|   044B  |       kn-IN      | Kannada - India                              |
|   044C  |       ml-IN      | Malayalam - India                            |
|   044D  |       as-IN      | Assamese - India                             |
|   044E  |       mr-IN      | Marathi - India                              |
|   044F  |       sa-IN      | Sanskrit - India                             |
|   0450  |       mn-MN      | Mongolian (Cyrillic) - Mongolia              |
|   0451  |       bo-CN      | Tibetan - China                              |
|   0452  |       cy-GB      | Welsh - United Kingdom                       |
|   0453  |       km-KH      | Khmer - Cambodia                             |
|   0454  |       lo-LA      | Lao - Lao PDR                                |
|   0456  |       gl-ES      | Galician - Spain                             |
|   0457  |      kok-IN      | Konkani - India                              |
|   0459  |    sd-Deva-IN    | (reserved) - (reserved)                      |
|   045A  |      syr-SY      | Syriac - Syria                               |
|   045B  |       si-LK      | Sinhala - Sri Lanka                          |
|   045C  |    chr-Cher-US   | Cherokee - Cherokee                          |
|   045D  |    iu-Cans-CA    | Inuktitut (Canadian_Syllabics) - Canada      |
|   045E  |       am-ET      | Amharic - Ethiopia                           |
|   0461  |       ne-NP      | Nepali - Nepal                               |
|   0462  |       fy-NL      | Frisian - Netherlands                        |
|   0463  |       ps-AF      | Pashto - Afghanistan                         |
|   0464  |      fil-PH      | Filipino - Philippines                       |
|   0465  |       dv-MV      | Divehi - Maldives                            |
|   0468  |    ha-Latn-NG    | Hausa - Nigeria                              |
|   046A  |       yo-NG      | Yoruba - Nigeria                             |
|   046B  |      quz-BO      | Quechua - Bolivia                            |
|   046C  |      nso-ZA      | Sesotho sa Leboa - South Africa              |
|   046D  |       ba-RU      | Bashkir - Russia                             |
|   046E  |       lb-LU      | Luxembourgish - Luxembourg                   |
|   046F  |       kl-GL      | Greenlandic - Greenland                      |
|   0470  |       ig-NG      | Igbo - Nigeria                               |
|   0473  |       ti-ET      | Tigrinya - Ethiopia                          |
|   0475  |      haw-US      | Hawiian - United States                      |
|   0478  |       ii-CN      | Yi - China                                   |
|   047A  |      arn-CL      | Mapudungun - Chile                           |
|   047C  |      moh-CA      | Mohawk - Canada                              |
|   047E  |       br-FR      | Breton - France                              |
|   0480  |       ug-CN      | Uyghur - China                               |
|   0481  |       mi-NZ      | Maori - New Zealand                          |
|   0482  |       oc-FR      | Occitan - France                             |
|   0483  |       co-FR      | Corsican - France                            |
|   0484  |      gsw-FR      | Alsatian - France                            |
|   0485  |      sah-RU      | Sakha - Russia                               |
|   0486  |    quc-Latn-GT   | K'iche - Guatemala                           |
|   0487  |       rw-RW      | Kinyarwanda - Rwanda                         |
|   0488  |       wo-SN      | Wolof - Senegal                              |
|   048C  |      prs-AF      | Dari - Afghanistan                           |
|   0491  |       gd-GB      | Scottish Gaelic - United Kingdom             |
|   0492  |    ku-Arab-IQ    | Central Kurdish - Iraq                       |
|   0801  |       ar-IQ      | Arabic - Iraq                                |
|   0803  |  ca-ES-valencia  | Valencian - Valencia                         |
|   0804  |       zh-CN      | Chinese (Simplified) - China                 |
|   0807  |       de-CH      | German - Switzerland                         |
|   0809  |       en-GB      | English - United Kingdom                     |
|   080A  |       es-MX      | Spanish - Mexico                             |
|   080C  |       fr-BE      | French - Belgium                             |
|   0810  |       it-CH      | Italian - Switzerland                        |
|   0813  |       nl-BE      | Dutch - Belgium                              |
|   0814  |       nn-NO      | Norwegian (Nynorsk) - Norway                 |
|   0816  |       pt-PT      | Portuguese - Portugal                        |
|   081A  |    sr-Latn-CS    | Serbian (Latin) - Serbia and Montenegro      |
|   081D  |       sv-FI      | Swedish - Finland                            |
|   0820  |       ur-IN      | Urdu - (reserved)                            |
|   082C  |    az-Cyrl-AZ    | Azeri (Cyrillic) - Azerbaijan                |
|   082E  |      dsb-DE      | Lower Sorbian - Germany                      |
|   0832  |       tn-BW      | Setswana / Tswana - Botswana                 |
|   083B  |       se-SE      | Sami (Northern) - Sweden                     |
|   083C  |       ga-IE      | Irish - Ireland                              |
|   083E  |       ms-BN      | Malay - Brunei Darassalam                    |
|   0843  |    uz-Cyrl-UZ    | Uzbek (Cyrillic) - Uzbekistan                |
|   0845  |       bn-BD      | Bangla - Bangladesh                          |
|   0846  |    pa-Arab-PK    | Punjabi - Pakistan                           |
|   0849  |       ta-LK      | Tamil - Sri Lanka                            |
|   0850  |    mn-Mong-CN    | Mongolian (Mong) - Mongolia                  |
|   0859  |    sd-Arab-PK    | Sindhi - Pakistan                            |
|   085D  |    iu-Latn-CA    | Inuktitut (Latin) - Canada                   |
|   085F  |    tzm-Latn-DZ   | Tamazight (Latin) - Algeria                  |
|   0867  |    ff-Latn-SN    | Pular - Senegal                              |
|   086B  |      quz-EC      | Quechua - Ecuador                            |
|   0873  |       ti-ER      | (reserved) - (reserved)                      |
|   0873  |       ti-ER      | Tigrinya - Eritrea                           |
|   0C01  |       ar-EG      | Arabic - Egypt                               |
|   0C04  |       zh-HK      | Chinese - Hong Kong SAR                      |
|   0C07  |       de-AT      | German - Austria                             |
|   0C09  |       en-AU      | English - Australia                          |
|   0C0A  |       es-ES      | Spanish - Spain                              |
|   0C0C  |       fr-CA      | French - Canada                              |
|   0C1A  |    sr-Cyrl-CS    | Serbian (Cyrillic) - Serbia and Montenegro   |
|   0C3B  |       se-FI      | Sami (Northern) - Finland                    |
|   0C6B  |      quz-PE      | Quechua - Peru                               |
|   1001  |       ar-LY      | Arabic - Libya                               |
|   1004  |       zh-SG      | Chinese - Singapore                          |
|   1007  |       de-LU      | German - Luxembourg                          |
|   1009  |       en-CA      | English - Canada                             |
|   100A  |       es-GT      | Spanish - Guatemala                          |
|   100C  |       fr-CH      | French - Switzerland                         |
|   101A  |       hr-BA      | Croatian (Latin) - Bosnia and Herzegovina    |
|   103B  |      smj-NO      | Sami (Lule) - Norway                         |
|   105F  |    tzm-Tfng-MA   | Central Atlas Tamazight (Tifinagh) - Morocco |
|   1401  |       ar-DZ      | Arabic - Algeria                             |
|   1404  |       zh-MO      | Chinese - Macao SAR                          |
|   1407  |       de-LI      | German - Liechtenstein                       |
|   1409  |       en-NZ      | English - New Zealand                        |
|   140A  |       es-CR      | Spanish - Costa Rica                         |
|   140C  |       fr-LU      | French - Luxembourg                          |
|   141A  |    bs-Latn-BA    | Bosnian (Latin) - Bosnia and Herzegovina     |
|   143B  |      smj-SE      | Sami (Lule) - Sweden                         |
|   1801  |       ar-MA      | Arabic - Morocco                             |
|   1809  |       en-IE      | English - Ireland                            |
|   180A  |       es-PA      | Spanish - Panama                             |
|   180C  |       fr-MC      | French - Monaco                              |
|   181A  |    sr-Latn-BA    | Serbian (Latin) - Bosnia and Herzegovina     |
|   183B  |      sma-NO      | Sami (Southern) - Norway                     |
|   1C01  |       ar-TN      | Arabic - Tunisia                             |
|   1c09  |       en-ZA      | English - South Africa                       |
|   1C0A  |       es-DO      | Spanish - Dominican Republic                 |
|   1C1A  |    sr-Cyrl-BA    | Serbian (Cyrillic) - Bosnia and Herzegovina  |
|   1C3B  |      sma-SE      | Sami (Southern) - Sweden                     |
|   2001  |       ar-OM      | Arabic - Oman                                |
|   2009  |       en-JM      | English - Jamaica                            |
|   200A  |       es-VE      | Spanish - Venezuela                          |
|   201A  |    bs-Cyrl-BA    | Bosnian (Cyrillic) - Bosnia and Herzegovina  |
|   203B  |      sms-FI      | Sami (Skolt) - Finland                       |
|   2401  |       ar-YE      | Arabic - Yemen                               |
|   2409  |      en-029      | English - Caribbean                          |
|   240A  |       es-CO      | Spanish - Colombia                           |
|   241A  |    sr-Latn-RS    | Serbian (Latin) - Serbia                     |
|   243B  |      smn-FI      | Sami (Inari) - Finland                       |
|   2801  |       ar-SY      | Arabic - Syria                               |
|   2809  |       en-BZ      | English - Belize                             |
|   280A  |       es-PE      | Spanish - Peru                               |
|   281A  |    sr-Cyrl-RS    | Serbian (Cyrillic) - Serbia                  |
|   2C01  |       ar-JO      | Arabic - Jordan                              |
|   2C09  |       en-TT      | English - Trinidad and Tobago                |
|   2C0A  |       es-AR      | Spanish - Argentina                          |
|   2C1A  |    sr-Latn-ME    | Serbian (Latin) - Montenegro                 |
|   3001  |       ar-LB      | Arabic - Lebanon                             |
|   3009  |       en-ZW      | English - Zimbabwe                           |
|   300A  |       es-EC      | Spanish - Ecuador                            |
|   301A  |    sr-Cyrl-ME    | Serbian (Cyrillic) - Montenegro              |
|   3401  |       ar-KW      | Arabic - Kuwait                              |
|   3409  |       en-PH      | English - Philippines                        |
|   340A  |       es-CL      | Spanish - Chile                              |
|   3801  |       ar-AE      | Arabic - U.A.E.                              |
|   380A  |       es-UY      | Spanish - Uruguay                            |
|   3C01  |       ar-BH      | Arabic - Bahrain                             |
|   3C0A  |       es-PY      | Spanish - Paraguay                           |
|   4001  |       ar-QA      | Arabic - Qatar                               |
|   4009  |       en-IN      | English - India                              |
|   400A  |       es-BO      | Spanish - Bolivia                            |
|   4409  |       en-MY      | English - Malaysia                           |
|   440A  |       es-SV      | Spanish - El Salvador                        |
|   4809  |       en-SG      | English - Singapore                          |
|   480A  |       es-HN      | Spanish - Honduras                           |
|   4C0A  |       es-NI      | Spanish - Nicaragua                          |
|   500A  |       es-PR      | Spanish - Puerto Rico                        |
|   540A  |       es-US      | Spanish - United States                      |
|   7C04  |      zh-CHT      | Chinese - Traditional                        |

</details>

**Comment by logounet** [comment]
Hello,

Here is the French translation of "Simple IP Config 2.9.4"
See attached "lang-fr-FR.json.zip" file.

[lang-fr-FR.zip](https://github.com/KurtisLiggett/Simple-IP-Config/files/8761992/lang-fr-FR.zip)


Best regards

**Reply by KurtisLiggett** [reply]
Thanks for the help! Added to the project.

**Comment by ms74ro** [comment]
Here is the Romanian translation for version 2.9.6.1
I added some omitted Romanian characters and modified an edit error. Sorry.

[Simple-IP-Config-2961-lang-ro-RO.zip](https://github.com/KurtisLiggett/Simple-IP-Config/files/8891210/Simple-IP-Config-2961-lang-ro-RO.zip)

**Reply by KurtisLiggett** [reply]
Thank you for your contribution! Minor formatting issues, but otherwise looks great.
Added to the repo.

**Comment by blackcrack** [comment]
@KurtisLiggett  ^^ :grinning: once over my post

**Reply by KurtisLiggett** [reply]
thanks, looks like I missed that one

**Comment by PvdE75** [comment]
Hello, i've made an Dutch NL translation.
See attached file

[lang-nl-NL.zip](https://github.com/KurtisLiggett/Simple-IP-Config/files/9168223/lang-nl-NL.zip)

**Reply by KurtisLiggett** [reply]
Thanks for the contribution!

**Comment by zahit1** [comment]
Hello, i've added Turkish tr-TR translation file.
[lang-tr-TR.zip](https://github.com/KurtisLiggett/Simple-IP-Config/files/11074842/lang-tr-TR.zip)

**Comment by andago9** [comment]
hello, I update the spanish language

[lang-es-ES.zip](https://github.com/KurtisLiggett/Simple-IP-Config/files/13044950/lang-es-ES.zip)

**Comment by Clementbln** [comment]
thanks
